main.go

```diff
@@ -254,6 +257,19 @@ func main() {
 		envString("", "GITSYNC_ASKPASS_URL", "GIT_SYNC_ASKPASS_URL", "GIT_ASKPASS_URL"),
 		"a URL to query for git credentials (username=<value> and password=<value>)")
 
+	flGithubAPIURL := pflag.String("github-api-url",
+		envString("api.github.com", "GITHUB_API_URL"),
+		"the API URL to use when making requests to GitHub when using app auth")
+	flGithubAppPrivateKey := pflag.String("github-app-private-key",
```

Comment:
We need to validate these (see ~ L493).  Specifically:

* You can't use these *and* some other form of auth at the same time
* If user sets this flag, they also need to set the app ID and installation ID

Reply:
Actually, we should not put private keys as flags, they are listable via /proc.  The --password flag is bad for the same reason. 

We should have a way to declare a non-flag which can be set by env but not via CLI.  I will think about that.  It's going to be a busy few weeks, so please hang tight.

 Also, we should be able to pass this in via a file, like `--password-file`.  For all of these, I'd like it to be possible to use a kubernetes secret and mount a volume, rather than flags/env vars.  I'll have to refresh myself on formatting, i think we can only demand one value per file.  So maybe we want an alternative `--github-app-config=<dir>` with documented files within it, e.g. `api-url` (optional), `private-key` (required), `application-id` (required), and `installation-id` (required).   That way a user could make a single k8s Secret object with those key, and use a Secret-type volume.

Reply:
> It's going to be a busy few weeks, so please hang tight.
No worries! thanks for taking the time for the detailed review, I will address the simple things you brought up up in the meantime

Reply:
+1 for using a single k8s Secret object with these keys and add validation for them. It would be much easier for the downstream service to provide these values.